### ISSUE TITLE
Offset pixel mobs don't reset position after attack animation.

### ISSUE THREAD

**Issue by SpadesNeil** [PolarisSS13/Polaris#2352]
#### Brief description of the issue
Large mobs, such as the Xeno Queen, should be centered on their tile by pixel_x = -16 upon being spawned. However, whenever they attack someone, or something, with melee, that offset gets immediately broken. (I know Xeno Queens attack with spit, I'm just using it as an example. Pretend it doesn't spit at you and claws your face off instead.)

#### What you expected to happen
After an attack animation, the mob should reset to its original position prior to going through the attack animation.

#### What actually happened
Its pixel_x variable resets to 0.

#### Steps to reproduce
1. Create a hostile mob with more than 32 pixels. Must be melee.
2. Code it so that it's offset by enough pixels to center it on the tile it's standing on.
3. Let it attack something.

#### Additional info:
 - **Server Revision**: https://github.com/PolarisSS13/Polaris/commit/729956de191474113e947f9f37ba56c83e6929e7
 - **Anything else you may wish to add**: This is not currently affecting anything I know of in-game, but it does render admin events useless if they involve spawned NPCs that happen to be big ones. It should be fixed when convenient to do so. I don't know where the attack animation is handled, or I'd do it myself. This affects all mobs as far as I can tell.

**Comment by Anewbe** [comment]
Wasn't supposed to close.

**Comment by Mechoid** [comment]
I asked about this in a PR from a while-ago, regarding this exact thing. There is a set of vars regarding the original pixel position, this is likely caused by them not being set, or being set to default as everything else is.

**Comment by SpadesNeil** [comment]
Yes, old_x and old_y are supposed to reset the positions, but these vars are getting ignored for some reason and I have been unable to determine the cause.

**Comment by SpadesNeil** [comment]
I can no longer reproduce this. Closing for now.

**Comment by SpadesNeil** [comment]
Re-opening because I discovered the problem.

New steps to reproduce:
1. Grab a large mob.
2. Release grab.
3. old_x and old_y are ignored.
